step-templates/roundhouse-database-migration.json

```diff
@@ -0,0 +1,257 @@
IsNullOrEmpty($roundhouseServerPort))\n        {\n            # Append to servername\n            $roundhouseServerName += \",$roundhouseServerPort\"\n\n            # Empty the port\n            $roundhouseServerPort = [string]::Empty\n        }\n    }\n    default\n    {\n        $roundhouseServerPort = \"Port=$roundhouseServerPort;\"\n    }\n}\n\n\n$roundhouseSwitches += \"--connectionstring=Server=$roundhouseServerName;$roundhouseServerPort $roundhouseUserInfo Database=$roundhouseDatabaseName;\"\n$roundhouseSwitches += \"--databasetype=$roundhouseDatabaseServerType\"\n$roundhouseSwitches += \"--silent\"\n\n# Check for folder definitions\nif (![string]::IsNullOrEmpty($roundhouseUpFolder))\n{\n    # Add up folder\n    $roundhouseSwitches += \"--up=$roundhouseUpFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseAlterDatabaseFolder))\n{\n    $roundhouseSwitches += \"--alterdatabasefolder=$roundhouseAlterDatabaseFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseRunBeforeUpFolder))\n{\n    $roundhouseSwitches += \"--runbeforeupfolder=$roundhouseRunBeforeUpFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseFunctionsFolder))\n{\n    $roundhouseSwitches += \"--functionsfolder=$roundhouseFunctionsFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseViewsFolder))\n{\n    $roundhouseSwitches += \"--viewsfolder=$roundhouseViewsFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseSprocsFolder))\n{\n    $roundhouseSwitches += \"--sprocsfolder=$roundhouseSprocsFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseIndexFolder))\n{\n    $roundhouseSwitches += \"--indexesfolder=$roundIndexesFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhouseRunAfterAnyTimeFolder))\n{\n    $roundhouseSwitches += \"--runAfterOtherAnyTimeScriptsFolder=$roundhouseRunAfterAnyTimeFolder\"\n}\n\nif (![string]::IsNullOrEmpty($roundhousePermissionsFolder))\n{\n    $roundhouseSwitches += \"--permissionsfolder=$roundhousePermissionsFolder\"\n}\n\nif ([System.Boolean]::Parse($roundhouseDryRun))\n{\n    $roundhouseSwitches += \"--dryrun\"\n}\n\nif ([System.Boolean]::Parse($roundhouseRecordOutput))\n{\n    $roundhouseSwitches += \"--outputpath=$roundhouseOutputPath\"\n}\n\n# Add transaction switch\n$roundhouseSwitches += \"--withtransaction=$($roundhouseWithTransaction.ToLower())\"\n\n# Check for version\nif (![string]::IsNullOrEmpty($roundhouseVersion))\n{\n    # Add version\n    $roundhouseSwitches += \"--version=$roundhouseVersion\"\n}\n\n# Execute RoundhousE\n& $roundhouseExecutable.FullName $roundhouseSwitches\n\n# If the output path was specified, attach artifacts\nif (![string]::IsNullOrEmpty($roundhouseOutputPath))\n{\n    # Get files that were generated\n    $artifacts = Get-ChildItem -Path $roundhouseOutputPath -Recurse | where { ! $_.PSIsContainer }\n\n    # Loop through results\n    ForEach($artifact in $artifacts)\n    {\n        # Attach deployment artifact\n        New-OctopusArtifact -Path \"$($artifact.FullName)\" -Name \"$($artifact.Name)\"\n    }\n}\n"
```

Comment:
What's the output of roundhouse look like? Is it potentially a lot of files? What would you think about zipping up the output folder and adding that as an artifact?

Would there ever be sensitive data in the output? We might need to add to the help text of the record output param that it adds artifacts and links to the docs on implications of that

Reply:
Added creating zip.  If there was sensitive data within the scripts themselves, then potentially it will since it will include the .SQL files executed with the output.